bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php

```diff
@@ -2488,4 +2488,168 @@ protected function processAvailableFieldDefinitions($fds, &$firstOne, &$commonFi
             }
         }
     }
+    
+    /**
+     * @Route("/export-json-config", methods={"GET"})
+     *
+     * @param Request $request
+     * 
+     * @return Response
+     */
+    public function exportJsonConfigAction(Request $request)
+    {
+        $importConfigId = $request->get('importConfigId');
+        
+        if($importConfigId == null || empty($importConfigId)){
+            throw new \Exception("Please provide a valid import configuration Id");
+        }
+
+        try {
+            $importConfig = ImportConfig::getById($importConfigId);
+        
+            $configData = $importConfig->getConfig();
+            $configName = $importConfig->getName();
+            
+            $response = new Response();
+
+            $response->setContent($configData);
+
+            $disposition = $response->headers->makeDisposition(
+                ResponseHeaderBag::DISPOSITION_ATTACHMENT,
+                $configName.".json"
+            );
+
+            $response->headers->set('Content-Type', 'application/json');
+            $response->headers->set('Content-Disposition', $disposition);
+
+            return $response;
+            
+        } catch (\Exception $e) {
+            throw new \Exception("Error retrieving import configuration - ".$e->getMessage());
+        }
+    }
+    
+    /**
+     * @Route("/import-json-config", methods={"POST"})
+     *
+     * @param Request $request
+     * @param ImportService $importService
+     *
+     * @return JsonResponse
+     */
+    public function importJsonConfigAction(Request $request, ImportService $importService)
+    {
+        $importConfigId = $request->get('importConfigId');
+        
+        $tmpName = $_FILES['Filedata']['tmp_name'];
+        $json = file_get_contents($tmpName);
+
+        $configData = json_decode($json, true);
+
+        $selectedGridColumns = $configData['selectedGridColumns'];
+        $resolverSettings = $configData['resolverSettings'];
+        $shareSettings = $configData['shareSettings'];
+        $dialect = json_decode(json_encode($configData['csvSettings']), false);
+        
+        $success = true;
+        $supportedFieldTypes = ['checkbox', 'country', 'date', 'datetime', 'href', 'image', 'input', 'language', 'table', 'multiselect', 'numeric', 'password', 'select', 'slider', 'textarea', 'wysiwyg', 'objects', 'multihref', 'geopoint', 'geopolygon', 'geopolyline', 'geobounds', 'link', 'user', 'email', 'gender', 'firstname', 'lastname', 'newsletterActive', 'newsletterConfirmed', 'countrymultiselect', 'objectsMetadata'];
```

Comment:
see question below but this shouldn't be hardcoded anyway. hard to maintain and even impossible to add custom datatypes.

Reply:
As said in the previous comment, this is part of the code of the importGetFileInfoAction. How can I modify to avoid this code duplication?

Reply:
well, importGetFileAction is about importing CSV data. Here we are talking about importing JSON configurations. Maybe take a look at 
https://github.com/pimcore/pimcore/blob/663b0887cf4623b5499987e2b15b9522f3342e11/bundles/AdminBundle/Controller/Admin/DataObject/ClassController.php#L460
for another sample as well.

I guess you don't need the whole magic.

Reply:
My idea was to use the imported json to refresh the CSV import interface the same way it's done when a saved configuration is loaded. I will follow your suggestion and try to make this more simple

Reply:
Thank you! 👍

Reply:
Made some changes.
As i wrote in the code's comments, I have maintained the reload of data form original CSV to properly refresh the data preview on the import interface